tests/db_functions/text/test_regexpreplace.py

```diff
@@ -0,0 +1,165 @@
+        else:
+            expression = RegexpReplace('name', Value(r'^(.*(?:R\. ?){2}) (.*)$'), Value(r'\2, \1'))
+        queryset = Author.objects.annotate(flipped_name=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 'Martin, George R. R.'),
+            ('J. R. R. Tolkien', 'Tolkien, J. R. R.'),
+        ], transform=lambda x: (x.name, x.flipped_name), ordered=False)
+
+    def test_expression(self):
+        expression = RegexpReplace(Concat(Value('Author: '), 'name'), Value(r'.*: '), Value(''))
+        queryset = Author.objects.annotate(same_name=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 'George R. R. Martin'),
+            ('J. R. R. Tolkien', 'J. R. R. Tolkien'),
+        ], transform=lambda x: (x.name, x.same_name), ordered=False)
+
+    def test_update(self):
+        Author.objects.update(name=RegexpReplace('name', Value(r'(R\. ){2}'), Value('')))
+        self.assertQuerysetEqual(Author.objects.all(), [
+            'George Martin',
+            'J. Tolkien',
+        ], transform=lambda x: x.name, ordered=False)
+
+    def test_first_occurrence(self):
+        expression = RegexpReplace('name', Value(r'R\. '), Value(''))
+        queryset = Author.objects.annotate(single_middlename=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 'George R. Martin'),
+            ('J. R. R. Tolkien', 'J. R. Tolkien'),
+        ], transform=lambda x: (x.name, x.single_middlename), ordered=False)
+
+
+class RegexpReplaceFlagTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        Article.objects.create(
+            title='Chapter One',
+            text='First Line.\nSecond Line.\nThird Line.',
+            written=Now(),
+        )
+
+    def test_global_flag(self):
+        # MariaDB only supports replacing all occurrences - test always passes.
+        expression = RegexpReplace('text', Value(r'Line'), Value('Word'), Value('g'))
+        article = Article.objects.annotate(result=expression).first()
+        self.assertEqual(article.result, 'First Word.\nSecond Word.\nThird Word.')
+
+    def test_dotall_flag(self):
+        expression = RegexpReplace('text', Value(r'\..'), Value(', '), Value('gs'))
+        article = Article.objects.annotate(result=expression).first()
+        self.assertEqual(article.result, 'First Line, Second Line, Third Line.')
+
+    def test_multiline_flag(self):
+        expression = RegexpReplace('text', Value(r' Line\.$'), Value(''), Value('gm'))
+        article = Article.objects.annotate(result=expression).first()
+        self.assertEqual(article.result, 'First\nSecond\nThird')
+
+    def test_extended_flag(self):
+        if connection.vendor == 'oracle':
+            # Oracle doesn't support comments in extended regular expressions.
+            pattern = Value(r"""
+                .
+                Line
+                \.
+            """)
```

Comment:
If it's valuable to test this case we should do this for all database and move a case with comments to a separate method (skipped on Oracle).